### PR TITLE
180 Better prints for executer failed and failed during wait and download

### PR DESCRIPTION
**Pull request by Pbarbosa92**
This PR addresses the issue [#180](https://github.com/inductiva/tasks/issues/180)

### CODE REVIEW

**Comment by Pbarbosa92** [comment]
How to trigger a executer failed
```
import inductiva

gromacs = inductiva.simulators.GROMACS()

input_dir = inductiva.utils.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "gromacs-input-example.zip",
    unzip=True,
)

commands = [
    "gmx solvate -cs tip4p -box 2.3 -o conf.gro -p topol.top",
    "gmx grompp -f energy_minimization.mdp -o min.tpr -pp min.top -po min.mdp -c conf.gro -p topol.top",
    "gmx mdrun -s min.tpr -o min.trr -c min.gro -e min.edr -g min.log",
    "gmx grompp -f positions_decorrelation.mdp -o decorr.tpr -pp decorr.top -po decorr.mdp -c min.gro",
    "gmx mdrun -s decorr.tpr -o decorr.trr -x  -c decorr.gro -e decorr.edr -g decorr.log",
    "gmx grompp -f simulation.mdp -o eql.tpr -pp eql.top -po eql.mdp -c decorr.gro",
    "asdgmx mdrun -s eql.tpr -o eql.trr -x trajectory.xtc -c eql.gro -e eql.edr -g eql.log",
]

task = gromacs.run(
    input_dir=input_dir,
    commands=commands,
)

task.wait()
task.download_outputs()
```

How to trigger a failed status
```
import inductiva

# Set simulation input directory
input_dir = inductiva.utils.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "openfast-input-example.zip", unzip=True)

# List of commands to run
commands = [
    "openfast hello_world.fst"
]

# Initialize OpenFAST simulator
openfast = inductiva.simulators.OpenFAST()

# Run simulation 
task = openfast.run(input_dir=input_dir,
                    commands=commands)

task.wait()
# task.download_outputs()

```

**Comment by Pbarbosa92** [comment]
**QA**
To QA trigger a failed and executer failed and check the wait() and task.download_outputs() prints.

**Comment by Franreno** [comment]
# QA

## Failed status 

### task.wait()

Task g46f3ugnfh9ik8x9pe07hwpzs failed.
Downloading stdout and stderr files to inductiva_output/g46f3ugnfh9ik8x9pe07hwpzs...
Partial download completed to inductiva_output/g46f3ugnfh9ik8x9pe07hwpzs.
Please inspect the stdout.txt and stderr.txt files at: inductiva_output/g46f3ugnfh9ik8x9pe07hwpzs
For more information.

### task.download_outputs()

Downloading simulation outputs to inductiva_output/9qom7uehkczyvchsmrp9z1vt7/output.zip...
100%|█████| 2.25M/2.25M [00:00<00:00, 4.78MB/s]
Uncompressing the outputs to inductiva_output/9qom7uehkczyvchsmrp9z1vt7...
Task 9qom7uehkczyvchsmrp9z1vt7 failed.
Please inspect the stdout.txt and stderr.txt files at: inductiva_output/9qom7uehkczyvchsmrp9z1vt7
For more information.

### Calling task.download_outputs() before task.wait()

Consider tracking the status of the task via CLI:
        inductiva tasks list --task-id rvcdp9c45ryti57ut43f7awxe
Or, tracking the logs of the task via CLI:
        inductiva logs rvcdp9c45ryti57ut43f7awxe
Error: Task output ZIP file does not exist.


### task killed

Task ryrtzwhtkq9p0dsu4xhr0iwpk has started and is now running remotely.
Task ryrtzwhtkq9p0dsu4xhr0iwpk killed.
Downloading stdout and stderr files to inductiva_output/ryrtzwhtkq9p0dsu4xhr0iwpk...
Partial download completed to inductiva_output/ryrtzwhtkq9p0dsu4xhr0iwpk.
Downloading simulation outputs to inductiva_output/ryrtzwhtkq9p0dsu4xhr0iwpk/output.zip...
100%|█| 2.25M/2.25M [00:00<00:00, 3.65MB/s]
Uncompressing the outputs to inductiva_output/ryrtzwhtkq9p0dsu4xhr0iwpk...

## Executer failed

Task q7k5sy893a9dpzdt7bdz973my has started and is now running remotely.
The remote process running the task failed:
 > Message: The command must start with 'gmx'.

## Forced zombie status

Task nhxbr8qd6xrp134eswtxaxx2f successfully queued and waiting to be picked-up for execution...
The task nhxbr8qd6xrp134eswtxaxx2f is about to start.
Task nhxbr8qd6xrp134eswtxaxx2f has started and is now running remotely.
An internal error occurred with status executer-terminated-by-user while performing the task. -> task.wait()
Error: Task output ZIP file does not exist. -> task.download_outputs()

--- 

Seems ok to me!

File: inductiva/tasks/task.py [hunk 1]
```diff
@@ -494,8 +496,14 @@ def wait(self,
...
                 if download_std_on_completion:
-                    self.download_outputs(filenames=self.STANDARD_OUTPUT_FILES)
-
+                    self._called_from_wait = True
+                    out_dir = self.download_outputs(
+                        filenames=self.STANDARD_OUTPUT_FILES)
+                    if status == models.TaskStatusCode.FAILED:
+                        logging.error(
+                            "Please inspect the stdout.txt and"
+                            " stderr.txt files at: %s\n"
```
Comment (sergiomsantos) [inline inductiva/tasks/task.py:505]:
```suggestion
                            " stderr.txt files at %s\n"
```

File: inductiva/tasks/task.py [hunk 2]
```diff
@@ -769,6 +797,12 @@ def download_outputs(
         if rm_remote_files:
             self.remove_remote_files()
 
+        if self._status == models.TaskStatusCode.FAILED:
+            logging.error(
+                "Task %s failed.\n"
+                "Please inspect the stdout.txt and stderr.txt files at: %s\n"
```
Comment (sergiomsantos) [inline inductiva/tasks/task.py:803]:
```suggestion
                "Please inspect the stdout.txt and stderr.txt files at %s\n"
```

**Comment by Pbarbosa92** [comment]
> just a minor typo correction

Done

**Review by sergiomsantos — APPROVED** [review]

**Pull request merged**